### PR TITLE
Add basic cluster framework

### PR DESCRIPTION
**Pull request by Xieql**
**What type of PR is this?**

/kind feature

**What this PR does / why we need it**:
Provides the basic framework of crd and controller for cluster

part  #96 

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

The new make script requires the project to be located in the GOPATH directory (default /root/go)

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -101,3 +110,15 @@ gen-check: gen
...
 serve:
 	hack/local-docsite-up.sh
+
+PHONY: init-codegen
+init-codegen:
+	hack/init-codegen.sh
+
+.PHONY: gen-crd
+gen-crd: init-codegen ## Generate WebhookConfiguration, ClusterRole and CustomResourceDefinition objects.
+	$(CONTROLLER_GEN) rbac:roleName=manager-role crd webhook paths="./..." output:crd:artifacts:config=manifests/crd/bases
```
Comment (zirain) [inline Makefile:120]: manifests/crd/bases -> manifests/charts/base/templates

make it configurable

File: hack/init-codegen.sh [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+if ! [ -x "$(command -v ginkgo)" ]; then
+    util::install_tools github.com/onsi/ginkgo/v2/ginkgo v2.0.0
+fi
+
+if ! [ -x "$(command -v controller-gen)" ]; then
+    util::install_tools sigs.k8s.io/controller-tools/cmd/controller-gen v0.8.0
+fi
+
+if ! [ -x "$(command -v deepcopy-gen)" ]; then
+    util::install_tools k8s.io/code-generator/cmd/deepcopy-gen latest
```
Comment (zirain) [inline hack/init-codegen.sh:30]: use definite version instead of `latest`

File: pkg/api/policy/v1alpha1/type.go [hunk 3]
```diff
@@ -0,0 +1,61 @@
+/*
```
Comment (zirain) [inline pkg/api/policy/v1alpha1/type.go:1]: TODO: revert this

Comment (zirain) [reply pkg/api/policy/v1alpha1/type.go:1]: rename to `pkg/apis`

Comment (zirain) [reply pkg/api/policy/v1alpha1/type.go:1]: lint should ignore `pkg/apis`, `pkg/client-go`

File: hack/init-codegen.sh [hunk 4]
```diff
@@ -0,0 +1,47 @@
+#!/usr/bin/env bash
```
Comment (hzxuzhonghu) [inline hack/init-codegen.sh:1]: If this is refereeced from some other place, please comment it and keep the original copyright

File: hack/init-codegen.sh (hunk 2 above)
Comment (hzxuzhonghu) [reply hack/init-codegen.sh:30]: Yes, and keep depending on same version

**Review by hzxuzhonghu — COMMENTED** [review]
Since it can work, please replace the stub with UCSCluster and CustomMachine

**Comment by Xieql** [comment]
> ok i will do this in the next pr after the current pr is done

**Comment by hzxuzhonghu** [comment]
I donot think we want a fake api to pollute main branch

**Comment by Xieql** [comment]
> I donot think we want a fake api to pollute main branch

Yes, I will remove the test type.go and generated files in the next commit

File: hack/init-codegen.sh (hunk 4 above)
Comment (Xieql) [reply hack/init-codegen.sh:1]: Not from some other places, like most other scripts, do not need to add copyright

**Comment by hzxuzhonghu** [comment]
please make the ci happy

File: Makefile (hunk 1 above)
Comment (Xieql) [reply Makefile]: completed

File: hack/init-codegen.sh (hunk 2 above)
Comment (Xieql) [reply hack/init-codegen.sh]: completed

File: pkg/api/policy/v1alpha1/type.go (hunk 3 above)
Comment (Xieql) [reply pkg/api/policy/v1alpha1/type.go]: completed

**Review by hzxuzhonghu — APPROVED** [review]
/lgtm

**Pull request merged**